printer/snmp.js

```diff
@@ -0,0 +1,151 @@
+const snmp = require('net-snmp');
+// see https://github.com/remetremet/SNMP-OIDs/blob/master/OIDs/Printer-HP-LaserJet-P2055.md
+const SNMP_OBJECT_IDS = {
+  TONER_CAPACITY: '1.3.6.1.2.1.43.11.1.1.8.1.1',
+  CURRENT_TONER_LEVEL: '1.3.6.1.2.1.43.11.1.1.9.1.1',
+  PAGES_PRINTED: '1.3.6.1.2.1.43.10.2.1.4.1.1',
+  DEVICE_UPTIME: '1.3.6.1.2.1.1.3.0',
+  MAC_ADDY: '1.3.6.1.2.1.2.2.1.6.1',
+  SERIAL_NUMBER: '1.3.6.1.2.1.43.5.1.1.17.1',
+  MODEL_NUMBER: '1.3.6.1.2.1.25.3.2.1.3.1',
+  MEMORY_SIZE: '1.3.6.1.2.1.25.2.3.1.5.1',
+  MEMORY_USED: '1.3.6.1.2.1.25.2.3.1.6.1'
+};
+class HPp2015dn {
+  constructor(printerIP) {
+    this.printerIP = printerIP;
+  }
+
+  /**
+   * inkLevel determines and returns the current ink level of a HP p2015dn printer
+   * @param {String} printerIP IP address of printer to query
+   * @returns Promise with current ink level of printer in % of capacity
+   */
+  async getInkLevel() {
+    const currentLevel = await this.getCurrentTonerLevel();
+    const capacity = await this.getTonerCapacity();
+
+    if (currentLevel && capacity) {
+      return (currentLevel / capacity) * 100;
+    }
+
+    return false;
+  }
+
+  /**
+   * getTonerCapacity makes a snmp query against toner capacity OID and returns value
+   * @returns Promise with toner capacity in unknown units
+   */
+  async  getTonerCapacity() {
+    return await this.executeSNMPRequest([SNMP_OBJECT_IDS.TONER_CAPACITY]);
+  }
+
+  /**
+   * getCurrentTonerLevel makes a snmp query against toner level OID and returns value
+   * @returns {Promise} Promise with current toner level in unknown units
+   */
+  async  getCurrentTonerLevel() {
+    return await this.executeSNMPRequest([
+      SNMP_OBJECT_IDS.CURRENT_TONER_LEVEL,
+    ]);
+  }
+  /**
+   * executreSNMPRequest makes a snmp query against the given OID
```

Comment:
update each of these comment blocks + below to include the actual function name and a short sentence of what it does.

it would also b so cool to include an example response e.g.

```
when the mac address is queried with SNMP, the printer returns a value of <thing you observed in your notes>
```